packages/mc-scripts/src/utils/get-config-diff.js

```diff
@@ -0,0 +1,269 @@
+const chalk = require('chalk');
+
+// Since not all terminal supports colors, to make things more consistent for testing purposes,
+// during tests the color used is appended before the string instead of coloring it.
+const isTest = process.env.NODE_ENV === 'test';
+const red = (str) => {
+  if (isTest) return 'red-' + str;
+  return chalk.red(str);
+};
+const green = (str) => {
+  if (isTest) return 'green-' + str;
+  return chalk.green(str);
+};
+
+const keysToSkip = new Set(['id']);
+
+const t = (indentLevel) => '  '.repeat(indentLevel);
+
+const getStringDiff = (oldStr, newStr, label, indentLevel = 0) => {
+  if (!oldStr && newStr) {
+    return `${t(indentLevel)}${label} added: ${green(newStr)}`;
+  } else if (oldStr && !newStr) {
+    return `${t(indentLevel)}${label} removed: ${red(oldStr)}`;
+  } else if (oldStr && newStr && oldStr !== newStr) {
+    return `${t(indentLevel)}${label} changed: ${red(oldStr)} => ${green(
+      newStr
+    )}`;
+  }
+};
+
+const getArrayDiff = (oldArray, newArray, indentLevel = 0) => {
+  const oldArraySet = new Set(oldArray);
+  const arrayDiff = [];
+
+  newArray?.forEach((item) => {
+    if (oldArraySet.has(item)) {
+      oldArraySet.delete(item);
+    } else {
+      arrayDiff.push(`${t(indentLevel)}Item with "${green(item)}" added`);
+    }
+  });
+  oldArraySet.forEach((item) => {
+    arrayDiff.push(`${t(indentLevel)}Item with "${red(item)}" removed`);
+  });
+
+  return arrayDiff.join('\n');
+};
+
+const getPermissionsDiff = (oldPermissions, newPermissions) => {
+  const permissionDiff = ['permissions changed'];
+
+  const mappedOldPermissions = oldPermissions.reduce(
+    (acc, { name, oAuthScopes }) => ({
```

Comment:
Nit: when using `reduce` we try to provide meaningful names to the variables, for example `acc` is not super clear what it refers to.